src/toxcore/net_crypto/crypto_connection.rs

```diff
@@ -27,6 +26,34 @@ pub const DEFAULT_RTT: u64 = 1000;
 /// rtt in milliseconds for TCP connections
 pub const TCP_RTT: u64 = 500;
 
+/// The dT for the average packet receiving rate calculations.
+pub const PACKET_COUNTER_AVERAGE_INTERVAL: u64 = 50;
+
+/// How many last sizes of `send_array` should be recorded for congestion
+/// control.
+pub const CONGESTION_QUEUE_ARRAY_SIZE: usize = 12;
+
+/// How many numbers of sent lossless packets should be recorded for congestion
+/// control. It should be bigger than `CONGESTION_QUEUE_ARRAY_SIZE` due to rtt.
+pub const CONGESTION_LAST_SENT_ARRAY_SIZE: usize = CONGESTION_QUEUE_ARRAY_SIZE * 2;
+
+/// Minimum packets rate per second.
+pub const CRYPTO_PACKET_MIN_RATE: f64 = 4.0;
```

Comment:
Why f64? Not f32. Does it need such large precision?

Reply:
Why would I choose f32? Is it performance or memory? I think the performance difference should be negligible, we can check later. Right now I just don't want to care about the precision so I use the higher type.